### ISSUE TITLE
Backbone, underscore and marionette

### ISSUE THREAD

**Issue by Olgagr** [tenex/rails-assets#114]
In my Gemfile I had:

```
gem 'rails-assets-backbone'
gem 'rails-assets-underscore'
gem 'rails-assets-backbone.marionette'
```

I did bundle install. I got post-install messages:

```
Post-install message from rails-assets-backbone.babysitter:
This component doesn't define main assets in bower.json.
Please open new pull request in component's repository:
https://github.com/marionettejs/backbone.babysitter
Post-install message from rails-assets-backbone.wreqr:
This component doesn't define main assets in bower.json.
Please open new pull request in component's repository:
https://github.com/marionettejs/backbone.wreqr
Post-install message from rails-assets-underscore:
This component doesn't define main assets in bower.json.
Please open new pull request in component's repository:
https://github.com/jashkenas/underscore
```

I added requirement messages to application.js : 

```
//= require backbone
//= require underscore
//= require backbone.marionette
```

When I open web inspector in Chrome I can see that files are included twice, even if some of them are blank:

```
<script src="/assets/backbone/backbone.js?body=1"></script>
<script src="/assets/backbone.js?body=1"></script>
<script src="/assets/underscore/underscore.js?body=1"></script>
<script src="/assets/underscore.js?body=1"></script>
<script src="/assets/backbone.marionette/backbone.marionette.js?body=1"></script>
<script src="/assets/backbone.marionette/core/amd/backbone.marionette.js?body=1"></script>
<script src="/assets/backbone.marionette.js?body=1"></script>
<script src="/assets/application.js?body=1"></script>
```

What am I missing ?

**Comment by porada** [comment]
If you `//= require package-from-rails-assets`, you’re actually requiring the manifest file of the gem, and the manifest file itself requires all files referenced under `main` in `bower.json` or all scripts inside the package if the `main` field is empty.

In case of `backbone`, `underscore` and `backbone.marionette`, those are respectively:

``` js
//= require backbone/backbone
```

``` js
//= require underscore/underscore
```

``` js
//= require backbone.marionette/backbone.marionette
//= require backbone.marionette/core/amd/backbone.marionette
```

If a manifest file doesn’t contain the `require_self` directive, it’s included as the last one—even if it’s empty (the `require` directives don’t count). That’s also why the empty `application.js` is referenced in the last `<script>` tag.

**tl;dr** This is Sprockets’ expected behavior, so it’s nothing to worry about. There will be no empty files in precompiled assets for production mode.

**Comment by Olgagr** [comment]
Ok, thanks for explanation.

**Comment by porada** [comment]
Please note that you can explicitly `//= require` e.g. `backbone.marionette/backbone.marionette` if you don’t need `backbone.marionette/core/amd/backbone.marionette`.
